app/wiring/AppComponents.scala

```diff
@@ -63,6 +66,15 @@ trait AppComponents extends PlayComponents with GzipFilterComponents {
   lazy val paypalController = wire[PaypalController]
   lazy val stripeController = new StripeController(paymentServices, stripeConfig)
 
+  // The in-scope variables crypto and csrfTokenSigner are both CSRFTokenSigner's,
+  // therefore, create a method which can be used to specify which one wire should use.
+  // Note: this will be unnecessary when crypto is removed in a future version of Play.
+  def getCSRFAddToken(crypto: CSRFTokenSigner): CSRFAddToken = wire[CSRFAddToken]
+  def getCSRFCheck(crypto: CSRFTokenSigner): CSRFCheck = wire[CSRFCheck]
+
+  lazy val addToken = getCSRFAddToken(csrfTokenSigner)
+  lazy val check = getCSRFCheck(crypto)
```

Comment:
As much as I like the `wire` macro, I think here it makes sense to directly instantiate the components like that:
```scala
lazy val csrfCheck: CSRFCheck = new CSRFCheck(csrfConfig, csrfTokenSigner)
lazy val csrfAddToken: CSRFAddToken = new CSRFAddToken(csrfConfig, csrfTokenSigner)
```

You could also put these csrf in the trait named `PlayComponents` as they are purely technical concerns